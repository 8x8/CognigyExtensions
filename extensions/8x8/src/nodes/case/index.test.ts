import { getCaseNodes } from './index';

describe('customer > getCaseNodes', () => {
  it('should create the nodes for case logic', () => {
    const nodes = getCaseNodes();

    expect(nodes).toMatchSnapshot();
  });
});
