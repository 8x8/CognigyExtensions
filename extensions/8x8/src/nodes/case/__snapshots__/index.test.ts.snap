// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer > getCaseNodes should create the nodes for case logic 1`] = `
[
  {
    "_id": undefined,
    "appearance": {
      "color": "#ff0050",
      "showIcon": true,
      "textColor": "black",
      "variant": "regular",
    },
    "behavior": {
      "entrypoint": false,
      "stopping": false,
    },
    "constraints": {
      "collapsable": false,
      "creatable": true,
      "deletable": true,
      "editable": true,
      "movable": true,
      "placement": {
        "children": {
          "whitelist": [],
        },
        "predecessor": {},
        "successor": {},
      },
    },
    "defaultLabel": "Get Case",
    "dependencies": {
      "children": [
        "onFoundCase",
        "onNotFoundCase",
      ],
    },
    "fields": [
      {
        "key": "connection",
        "label": "8x8 CRM Connection",
        "params": {
          "connectionType": "eightbyeight",
          "required": true,
        },
        "type": "connection",
      },
      {
        "description": "Filter the case by case number. Maximum 30 characters",
        "key": "filter$caseNum",
        "label": "Case Number",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by account number. Maximum 30 characters",
        "key": "filter$accountNum",
        "label": "Account Number",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by last name of customer. Maximum 30 characters. Supports partial strings",
        "key": "filter$lastName",
        "label": "Last name of customer",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by company. Maximum 50 characters. Supports partial strings",
        "key": "filter$company",
        "label": "Company",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by case status. Maximum 30 characters",
        "key": "filter$status",
        "label": "Case Status",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by project. Maximum 30 characters",
        "key": "filter$project",
        "label": "Project",
        "type": "cognigyText",
      },
      {
        "description": "Filter the case by case subject. Maximum 50 characters. Supports partial strings",
        "key": "filter$subject",
        "label": "Subject",
        "type": "cognigyText",
      },
      {
        "defaultValue": {},
        "description": "Custom fields and their parameter names are added by the administrator of the tenant.",
        "key": "filter$customFields",
        "label": "Custom Fields",
        "type": "json",
      },
      {
        "defaultValue": "input",
        "key": "storeLocation",
        "label": "Where to store the result",
        "params": {
          "options": [
            {
              "label": "Input",
              "value": "input",
            },
            {
              "label": "Context",
              "value": "context",
            },
          ],
          "required": true,
        },
        "type": "select",
      },
      {
        "condition": {
          "key": "storeLocation",
          "value": "input",
        },
        "defaultValue": "case",
        "key": "inputKey",
        "label": "Input Key to store Result",
        "type": "cognigyText",
      },
      {
        "condition": {
          "key": "storeLocation",
          "value": "context",
        },
        "defaultValue": "case",
        "key": "contextKey",
        "label": {
          "default": "Context Key to store Result",
        },
        "type": "cognigyText",
      },
    ],
    "form": [
      {
        "key": "connection",
        "type": "field",
      },
      {
        "key": "filter$caseNum",
        "type": "field",
      },
      {
        "key": "filter$accountNum",
        "type": "field",
      },
      {
        "key": "filter$lastName",
        "type": "field",
      },
      {
        "key": "filter$company",
        "type": "field",
      },
      {
        "key": "filter$status",
        "type": "field",
      },
      {
        "key": "filter$project",
        "type": "field",
      },
      {
        "key": "filter$subject",
        "type": "field",
      },
      {
        "key": "filter$customFields",
        "type": "field",
      },
      {
        "key": "storage",
        "type": "section",
      },
    ],
    "function": [Function],
    "parentType": null,
    "preview": undefined,
    "sections": [
      {
        "defaultCollapsed": true,
        "fields": [
          "storeLocation",
          "inputKey",
          "contextKey",
        ],
        "key": "storage",
        "label": "Storage Option",
      },
    ],
    "summary": "Retrieves case details from 8x8 CRM",
    "tags": [],
    "tokens": [],
    "type": "getCase",
  },
  {
    "_id": undefined,
    "appearance": {
      "color": "#61d188",
      "showIcon": true,
      "textColor": "white",
      "variant": "mini",
    },
    "behavior": {
      "entrypoint": false,
      "stopping": false,
    },
    "constraints": {
      "collapsable": false,
      "creatable": false,
      "deletable": false,
      "editable": false,
      "movable": false,
      "placement": {
        "children": {
          "whitelist": [],
        },
        "predecessor": {
          "whitelist": [],
        },
        "successor": {},
      },
    },
    "defaultLabel": "Case found",
    "dependencies": {
      "children": [],
    },
    "fields": [],
    "form": [],
    "function": null,
    "parentType": "getCase",
    "preview": undefined,
    "sections": [],
    "summary": "",
    "tags": [],
    "tokens": [],
    "type": "onFoundCase",
  },
  {
    "_id": undefined,
    "appearance": {
      "color": "#61d188",
      "showIcon": true,
      "textColor": "white",
      "variant": "mini",
    },
    "behavior": {
      "entrypoint": false,
      "stopping": false,
    },
    "constraints": {
      "collapsable": false,
      "creatable": false,
      "deletable": false,
      "editable": false,
      "movable": false,
      "placement": {
        "children": {
          "whitelist": [],
        },
        "predecessor": {
          "whitelist": [],
        },
        "successor": {},
      },
    },
    "defaultLabel": "Case not found",
    "dependencies": {
      "children": [],
    },
    "fields": [],
    "form": [],
    "function": null,
    "parentType": "getCase",
    "preview": undefined,
    "sections": [],
    "summary": "",
    "tags": [],
    "tokens": [],
    "type": "onNotFoundCase",
  },
]
`;
